app/core/AppStateManager.test.ts

```diff
@@ -0,0 +1,152 @@
+    });
+    (AppState.addEventListener as jest.Mock).mockImplementation((_, listener) => {
+      mockAppStateListener = listener;
+      return { remove: jest.fn() };
+    });
+    appStateManager = new AppStateManager();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  it('should subscribe to AppState changes on instantiation', () => {
+    expect(AppState.addEventListener).toHaveBeenCalledWith('change', expect.any(Function));
+  });
+
+  it('should track event when app becomes active and conditions are met', () => {
+    (store.getState as jest.Mock).mockReturnValue({
+      security: { dataCollectionForMarketing: true },
+    });
+    (extractURLParams as jest.Mock).mockReturnValue({ params: { attributionId: 'test123' } });
+
+    appStateManager.setCurrentDeeplink('metamask://connect?attributionId=test123');
+    mockAppStateListener('active');
+    jest.advanceTimersByTime(2000);
+
+    expect(mockTrackEvent).toHaveBeenCalledWith(
+      MetaMetricsEvents.APP_OPENED,
+      { attributionId: 'test123' },
+      true
+    );
+  });
+
+  it('should not track event when data collection is disabled', () => {
+    (store.getState as jest.Mock).mockReturnValue({
+      security: { dataCollectionForMarketing: false },
+    });
+
+    mockAppStateListener('active');
+    jest.advanceTimersByTime(2000);
+
+    expect(mockTrackEvent).toHaveBeenCalledWith(
+      MetaMetricsEvents.APP_OPENED,
+      { attributionId: undefined },
+      true
+    );
+  });
+
+  it('should not track event when there is no deeplink', () => {
+    (store.getState as jest.Mock).mockReturnValue({
+      security: { dataCollectionForMarketing: true },
+    });
+
+    mockAppStateListener('active');
+    jest.advanceTimersByTime(2000);
+
+    expect(mockTrackEvent).toHaveBeenCalledWith(
+      MetaMetricsEvents.APP_OPENED,
+      { attributionId: undefined },
+      true
+    );
+  });
+
+  it('should handle errors gracefully', () => {
+    (store.getState as jest.Mock).mockImplementation(() => {
+      throw new Error('Test error');
+    });
+
+    mockAppStateListener('active');
+    jest.advanceTimersByTime(2000);
+
+    expect(Logger.error).toHaveBeenCalledWith(
+      expect.any(Error),
+      'AppStateManager: Error processing app state change'
+    );
+    expect(mockTrackEvent).not.toHaveBeenCalled();
+  });
+
+  it('should clean up the AppState listener on cleanup', () => {
+    const mockRemove = jest.fn();
+    (AppState.addEventListener as jest.Mock).mockReturnValue({ remove: mockRemove });
+
+    appStateManager = new AppStateManager();
+    appStateManager.cleanup();
+
+    expect(mockRemove).toHaveBeenCalled();
+  });
+
+  it('should not process app state change when app is not becoming active', () => {
+    mockAppStateListener('background');
+    jest.advanceTimersByTime(2000);
+
+    expect(mockTrackEvent).not.toHaveBeenCalled();
+  });
+
+  it('should not process app state change when app state has not changed', () => {
```

Comment:
```suggestion
  it('does not process app state change when app state is the same', () => {
```
Q: Does this also work for other states?

Reply:
we dont want to send the same event twice

Reply:
I get it. Does this test also work if the state goes from background to background? Here we only test active.